OpenRA.Mods.Common/Activities/Air/ResupplyAircraft.cs

```diff
@@ -48,12 +50,23 @@ public override Activity Tick(Actor self)
 				}
 				else
 				{
-					// Helicopters should take off from their helipad immediately after resupplying.
-					// HACK: Append NextActivity to TakeOff to avoid moving to the Rallypoint (if NextActivity is non-null).
-					inner = ActivityUtils.SequenceActivities(
-						aircraft.GetResupplyActivities(host)
-						.Append(new AllowYieldingReservation(self))
-						.Append(new TakeOff(self)).Append(NextActivity).ToArray());
+                    // Helicopters should take off from their helipad immediately after resupplying,
+                    // unless the "Enter" or "Return to Base" command was explictly given by user
```

Comment:
And this is where your logic fails: just because the player selected the the dock manually - just because he has more docks and wants to micro where each heli should resupply - doesn't mean the heli should stay there.